src/parser.ts

```diff
@@ -662,6 +672,165 @@ const TRAVERSE_BUILDER: MarkVisitor<(rhs: TraversalResult | null) => TraversalRe
   },
 }
 
+const SELECTOR_BUILDER: MarkVisitor<null> = {
+  square_bracket(p) {
+    p.process(TRAVERSE_BUILDER)
+    return null
+  },
+
+  slice(p) {
+    p.process(TRAVERSE_BUILDER)
+    return null
+  },
+
+  group(p) {
+    // This should only throw an error until we add support for groups in selectors.
+    throw new Error('Invalid selector syntax')
+  },
+
+  everything() {
+    throw new Error('Invalid selector syntax')
+  },
+
+  this() {
+    throw new Error('Invalid selector syntax')
+  },
+
+  parent() {
+    throw new Error('Invalid selector syntax')
+  },
+
+  dblparent(p) {
+    throw new Error('Invalid selector syntax')
+  },
+
+  traverse(p) {
+    // p.unshift()
+    const _ = p.process(EXPR_BUILDER)
+    p.process(TRAVERSE_BUILDER)
+    return null
+  },
+
+  this_attr(p) {
+    p.processString()
+    return null
+  },
+
+  attr_access(p) {
+    p.processString()
+    return null
+  },
+
```

Comment:
We don't need this here. `attr_access` is only valid inside the traversal-context and we only use `SELECTOR_BUILDER` for expressions. I can look into improving the typing here.

```suggestion
```